cpp-package/src/board_shim.cpp

```diff
@@ -168,6 +168,29 @@ BrainFlowArray<double, 2> BoardShim::get_board_data ()
     delete[] buf;
     return matrix;
 }
+BrainFlowArray<double, 2> BoardShim::get_board_data (int num_datapoints)
+{
+
+    if (num_datapoints <= 0)
+    {
+        throw std::invalid_argument (
```

Comment:
change that to BrainFlowException and the second argument should be `(int)BrainFlowExitCodes::INVALID_ARGUMENTS_ERROR`